### PR TITLE
Order aggregate refactoring

### PR DESCRIPTION
**Pull request by vladislav-kozachenko**
In this PR we have refactored classes that relate to the order aggregate as follows:
- Removed unnecessary whitespaces
- Added Javadocs for all public methods and classes
- Extracted variables in calls
- Extracted returning values
- Split long methods
- Validated null arguments in public static methods

### CODE REVIEW

File: api-java/src/main/java/javaclasses/mealorder/c/order/OrderAggregate.java [hunk 1]
```diff
@@ -248,6 +223,16 @@ void orderCanceled(OrderCanceled event) {
...
     }
 
+    @Apply
+    void orderProcessed(OrderProcessed event) {
+        getBuilder().setStatus(ORDER_PROCESSED)
+                    .build();
+    }
+
+    /*
+     * Reacts
```
Comment (armiol) [inline api-java/src/main/java/javaclasses/mealorder/c/order/OrderAggregate.java]: @vladislav-kozachenko we say "Reactions".

File: api-java/src/main/java/javaclasses/mealorder/c/order/OrderRepository.java [hunk 2]
```diff
@@ -40,27 +40,45 @@
...
+            final PurchaseOrderCanceled purchaseOrderCanceled = (PurchaseOrderCanceled) message;
+            final Set<OrderId> orderIds =
+                    filterOrderIds(purchaseOrderCanceled.getOrderList());
+            return orderIds;
         });
     }
+
+    /**
+     * Extracts {@code OrderId} values from list of orders.
+     * Checks each order whether it is in repository.
```
Comment (YuriiHaidamaka) [inline api-java/src/main/java/javaclasses/mealorder/c/order/OrderRepository.java]: Reformat this Javadoc.

File: api-java/src/main/java/javaclasses/mealorder/c/order/Orders.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+     * Checks whether the menu is available on the date of the order.
+     *
+     * @param range     menu date range to check
+     * @param orderDate order date to check
+     * @return boolean true if there is a menu on the order date
+     */
+    public static boolean checkRangeIncludesDate(MenuDateRange range, LocalDate orderDate) {
+        checkNotNull(range);
+        checkNotNull(orderDate);
+        Comparator<LocalDate> comparator = new LocalDateComparator();
```
Comment (YuriiHaidamaka) [inline api-java/src/main/java/javaclasses/mealorder/c/order/Orders.java]: final

File: api-java/src/test/java/javaclasses/mealorder/c/order/AddDishToOrderTest.java [hunk 4]
```diff
@@ -63,28 +62,30 @@
...
+        final CreateOrder createOrder = createOrderInstance();
         final Command createOrderCommand = requestFactory.command()
-                                                         .create(toMessage(createOrder));
+                                                         .create(createOrder);
         commandBus.post(createOrderCommand, StreamObservers.noOpObserver());
     }
 
     @Test
     @DisplayName("produce DishAddedToOrder event")
     void produceEvent() {
```
Comment (YuriiHaidamaka) [inline api-java/src/test/java/javaclasses/mealorder/c/order/AddDishToOrderTest.java]: Delete this line.

File: api-java/src/test/java/javaclasses/mealorder/c/order/AddDishToOrderTest.java [hunk 5]
```diff
@@ -94,10 +95,10 @@ void produceEvent() {
     @DisplayName("add dish to order")
     void addDishToOrder() {
```
Comment (YuriiHaidamaka) [inline api-java/src/test/java/javaclasses/mealorder/c/order/AddDishToOrderTest.java]: Delete this line.

File: api-java/src/test/java/javaclasses/mealorder/c/order/AddDishToOrderTest.java [hunk 6]
```diff
@@ -121,30 +122,25 @@ void addDishToOrder() {
     void notAddDish() {
```
Comment (YuriiHaidamaka) [inline api-java/src/test/java/javaclasses/mealorder/c/order/AddDishToOrderTest.java]: Delete this line.

File: api-java/src/test/java/javaclasses/mealorder/c/order/AddDishToOrderTest.java [hunk 7]
```diff
@@ -153,32 +149,26 @@ void notAddDish() {
     @DisplayName("throw CannotAddDishToNotActiveOrder rejection")
     void notAddDishToNotActiveOrder() {
```
Comment (YuriiHaidamaka) [inline api-java/src/test/java/javaclasses/mealorder/c/order/AddDishToOrderTest.java]: Delete this line.

File: api-java/src/test/java/javaclasses/mealorder/c/order/CancelOrderTest.java [hunk 8]
```diff
@@ -81,10 +80,10 @@ public void setUp() {
     @DisplayName("produce OrderCanceled event")
     void produceEvent() {
```
Comment (YuriiHaidamaka) [inline api-java/src/test/java/javaclasses/mealorder/c/order/CancelOrderTest.java]: Delete this line.

File: api-java/src/test/java/javaclasses/mealorder/c/order/CancelOrderTest.java [hunk 9]
```diff
@@ -102,10 +101,10 @@ void produceEvent() {
     @DisplayName("cancel order")
     void cancelOrder() {
```
Comment (YuriiHaidamaka) [inline api-java/src/test/java/javaclasses/mealorder/c/order/CancelOrderTest.java]: Delete this line.

File: api-java/src/test/java/javaclasses/mealorder/c/order/CancelOrderTest.java [hunk 10]
```diff
@@ -140,7 +139,7 @@ void testOrderCanceledByReact() {
                                                             .setOrderId(ORDER_ID)
```
Comment (YuriiHaidamaka) [inline api-java/src/test/java/javaclasses/mealorder/c/order/CancelOrderTest.java]: Delete 136 this line.

File: api-java/src/test/java/javaclasses/mealorder/c/order/CancelOrderTest.java [hunk 11]
```diff
@@ -177,7 +176,7 @@ void throwsCannotCancelProcessedOrder() {
...
```
Comment (YuriiHaidamaka) [inline api-java/src/test/java/javaclasses/mealorder/c/order/CancelOrderTest.java]: Delete 173 line.

File: api-java/src/test/java/javaclasses/mealorder/c/order/CreateOrderTest.java [hunk 12]
```diff
@@ -80,10 +79,10 @@ public void setUp() {
     @DisplayName("produce OrderCreated event")
     void produceEvent() {
```
Comment (YuriiHaidamaka) [inline api-java/src/test/java/javaclasses/mealorder/c/order/CreateOrderTest.java]: Delete this line.

File: api-java/src/test/java/javaclasses/mealorder/c/order/CreateOrderTest.java [hunk 13]
```diff
@@ -136,15 +135,15 @@ void createOrder() {
     @DisplayName("create the order after its cancellation")
     void createOrderAfterCancellation() {
```
Comment (YuriiHaidamaka) [inline api-java/src/test/java/javaclasses/mealorder/c/order/CreateOrderTest.java]: Delete this line.

File: api-java/src/test/java/javaclasses/mealorder/c/order/CreateOrderTest.java [hunk 14]
```diff
@@ -176,10 +175,10 @@ void createOrderAfterCancellation() {
     @DisplayName("throw OrderAlreadyExists rejection")
     void notCreateOrder() {
```
Comment (YuriiHaidamaka) [inline api-java/src/test/java/javaclasses/mealorder/c/order/CreateOrderTest.java]: Delete this line.

File: api-java/src/test/java/javaclasses/mealorder/c/order/CreateOrderTest.java [hunk 15]
```diff
@@ -205,7 +204,7 @@ void throwMenuNotAvailableIfMenuAbsent() {
```
Comment (YuriiHaidamaka) [inline api-java/src/test/java/javaclasses/mealorder/c/order/CreateOrderTest.java]: Delete this line.

File: api-java/src/test/java/javaclasses/mealorder/c/order/CreateOrderTest.java [hunk 16]
```diff
@@ -259,7 +258,7 @@ void throwMenuNotAvailableIfVendorAbsent() {
```
Comment (YuriiHaidamaka) [inline api-java/src/test/java/javaclasses/mealorder/c/order/CreateOrderTest.java]: Delete this line.

File: api-java/src/main/java/javaclasses/mealorder/c/order/OrderAggregateRejections.java [hunk 17]
```diff
@@ -72,47 +100,90 @@ public static void throwMenuNotAvailable(CreateOrder cmd) throws MenuNotAvailabl
         }
```
[the remark this answers is not in the document]
Comment (vladislav-kozachenko) [reply api-java/src/main/java/javaclasses/mealorder/c/order/OrderAggregateRejections.java]: I think in this case it is unnecessarily. It's not a return statement.

**Pull request merged**